### PR TITLE
New kotlin parser

### LINKED ISSUE #1643: Remove or mention that kotlin parser is slow.
I came here because I can't open an issue in [their repository](https://github.com/tormodatt/tree-sitter-kotlin) which is suspicious!!
I know that Kotlin has dozens of features which makes parsing it harder.
So I did some investigations and I found out that the Kotlin's parser.c is huge ([92M before compiling](https://github.com/tormodatt/tree-sitter-kotlin/blob/master/src/parser.c), 18M after compiling) , which means 
40 bigger than Bash
80 bigger than Js
It takes more than 10s seconds to open every Kotlin file even with files that contain only one line .

So I believe that it's not usable or at least slow. so I think you guys need to mention that in Readme or just remove it

### PR DESCRIPTION
**Pull request by Joakker**
This PR replaces the old ~~stinky~~ kotlin parser for a new and improved one, with emphasis on simplicity and performance.

Hopefully fixes #1643

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -194,13 +194,13 @@ list.java = {
...
 
-list.kotlin = {
+--[[ list.kotlin = {
   install_info = {
     url = "https://github.com/tormodatt/tree-sitter-kotlin",
     files = { "src/parser.c" },
   },
   maintainers = { "@tormodatt" },
-}
+} ]]
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:203]: Is there already a link for the new kotlin parser?

Comment (Joakker) [reply lua/nvim-treesitter/parsers.lua:203]: It's [here](https://github.com/Joakker/tree-sitter-kotlin), though it's definitely not ready yet

**Comment by Joakker** [comment]
I don't understand what's wrong. I specified the link to the grammar so it should have installed it. @theHamsta help?

**Comment by stsewd** [comment]
You probably need to change the lockfile https://github.com/nvim-treesitter/nvim-treesitter/blob/6cf0a6692d5cf81f5d2480c0435a163a7dc335d1/lockfile.json#L113-L115

**Comment by Joakker** [comment]
Thanks, at least now the failing tests are not my fault

**Comment by theHamsta** [comment]
There are still some failure cases
image


- `package` is recognized as an identifier
- block comments are not parsed correctly
- inline, fun  

e.g. https://github.com/fwcd/kotlin-debug-adapter/blob/eb8f4bc2f86818c84f787fa91c21f62c9ccf8b31/adapter/src/main/kotlin/org/javacs/ktda/util/Utils.kt#L14

I really recommend running your parser on large repo and try to fix all error cases `tree-sitter parser **/*.kt` (when your bash supports `**` (else use find or just *.kt)

**Comment by Joakker** [comment]
Will do 👍🏽

**Comment by theHamsta** [comment]
@Joakker but startup time are already better now

**Comment by theHamsta** [comment]
@Joakker I see that you're working on the Kotlin parser. I think this is ready to merge! People will report issues to you.

Fix #1643.

**Review by theHamsta — APPROVED** [review]

**Comment by Joakker** [comment]
yey-evil

**Comment by tormodatt** [comment]
I tested a few files, and I'd say there are still more errors in this parser than in the old one, but the startuptime in the old one made it almost unusable. this version is much better, and if @Joakker  is motiviated to maintain this I think it can get very good! 🥇 

let we know you want me start reporting issues to your parser @Joakker , or if you have enough on your list for the time being

**Comment by theHamsta** [comment]
But I suspect that there is an issue with Tree-Sitter that increased the query load time from 300ms to 3s when upgrading to 0.19.0. Maybe perf can find out what Tree-Sitter is doing while loading the old Tree-Sitter-kotlin.

**Comment by Joakker** [comment]
@tormodatt feel free to report issues to me 😃

**Pull request merged**